### DISCUSSION TITLE
Are old blocks deleted when not connected to an element?

### DISCUSSION THREAD

**Discussion by lassemt** [verbb/super-table#470] [Q&A]
Are the old super table blocks deleted from the database when updating a super table field like this?
```
$field = Craft::$app->getFields()->getFieldByHandle('accordions');
$blockTypes = SuperTable::$plugin->getService()->getBlockTypesByFieldId($field->id);
$blockType = $blockTypes[0]; // There will only ever be one SuperTable_BlockType

$superTableData = [];
foreach ($accordions as $key => $accordion) {
  $superTableData['row' . $key] = [
    'type' => $blockType->id,
    'enabled' => true,
    'fields' => [
      'heading' => $accordion->heading,
      'text' => $accordion->text
    ]
  ];
}
$entry->setFieldvalue('accordions', $superTableData);
```
If not, what is the best way to remove old blocks that are not used anymore?

**Comment by engram-design** [comment]
It depends what the `$accordions` variable is, as you haven't specified what that is. So the answer is no, they aren't deleted, which is the same case for Matrix.

You'll want to be wary of deleting blocks that are attached to previous versions of an entry however - if your section is enabled with versions.

If you wanted to delete them, you'll have to loop through existing blocks and delete them, but that's a bit of a waste of resources unless you really *need* to delete them. If their content hasn't changed, passing the in block ID as the key would be more efficient.

```php
$blocks = $entry->accordions->all();

foreach ($blocks as $key => $accordion) {
    // ...
}

foreach ($blocks as $block) {
    Craft::$app->getElements()->deleteElement($block);
}
```

By the way, you should use `'new' . $key` and now `row`.
